### PR TITLE
refactor: #CRRE-525 refactor admin interface front

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
Refactoring of the admin interface

## Checklist tests
Ability to refuse/accept projects.
Being able to send orders to the bookseller.
Do the same for a selection and with everything selected (without loading all the pages)
The filters are working properly.
On the admin page, in the list of other filters, there is no longer the license filter.

## Issue ticket number and link
[CRRE-525](https://jira.support-ent.fr/browse/CRRE-525)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/template/administrator/order/filters.html [hunk 1]
```diff
@@ -3,58 +3,58 @@ <h1><i18n>crre.filters</i18n></h1>
...
+<!--        <span style="line-height: 50px;"><i18n>crre.licences.types</i18n></span>-->
+<!--        <multi-combo title="[[lang.translate('crre.all.licences')]]"-->
+<!--                     combo-model="licences"-->
+<!--                     search-on="name" order-by="name"-->
+<!--                     labels="comboLabels" filtered-model="filterChoice.licence"-->
+<!--                     selection-event="vm.applyFilter()"-->
+<!--                     deselection-event="vm.applyFilter()"-->
+<!--                     class="inline-block eight right-magnet">-->
+<!--        </multi-combo>-->
+<!--    </div>-->
```
Comment (BastienBenard) [inline src/main/resources/public/template/administrator/order/filters.html:59]: Partie utilisé pour les licences, on garde quand même pour le moment

File: src/main/resources/public/template/administrator/order/order-waiting.html [hunk 2]
```diff
@@ -1,216 +1,140 @@
-<div ng-controller="orderRegionController">
```
Comment (BastienBenard) [inline src/main/resources/public/template/administrator/order/order-waiting.html:1]: Dans ce fichier j'ai enlevé ce contrôleur ...

File: src/main/resources/public/template/administrator/order/order-waiting.html [hunk 3]
```diff
@@ -1,216 +1,140 @@
...
-            <div class="cell one margin-top-14 twelve-mobile" ng-click="openFiltersLightbox()">
-                <span class="color-red" style="text-decoration: underline;"><i18n>crre.more.filters</i18n></span>
-            </div>
-            <div class="cell six twelve-mobile" ng-if="filters.all.length > 0">
-                <div class="remove-margin-bottom">
-                    <div ng-repeat="key in filterChoiceCorrelation.keys" class="select-blocks inline-block">
-                        <div ng-repeat="item in filterChoice[key]"
-                             ng-click="dropElement(item,key)">
-                            [[item.toString()]]
-                        </div>
```
Comment (BastienBenard) [inline src/main/resources/public/template/administrator/order/order-waiting.html:76]: ce bloque de code est partie dans la directive project-filter-admin

File: src/main/resources/public/template/administrator/order/order-waiting.html [hunk 4]
```diff
@@ -1,216 +1,140 @@
...
+                        <div class="NEW status-circle  cell row myOrder centerCircleMobile"></div>
+                        <span class="zero-mobile"><i18n>crre.new</i18n></span>
                     </div>
-                </div>
-            </div>
-            <div class="cell twelve-mobile right-magnet">
-                <button ng-click="openConfirmGenerateLibraryLightbox()" class="right-magnet" ng-disabled="display.loading">
-                    <i18n>crre.take.order.library</i18n>
-                </button>
-            </div>
```
Comment (BastienBenard) [inline src/main/resources/public/template/administrator/order/order-waiting.html:84]: ce bloque de code est partie dans la directive project-filter-admin

File: src/main/resources/public/ts/controllers/administrator/order/waitingOrder.ts [hunk 5]
```diff
@@ -65,35 +56,22 @@ export const waitingOrderRegionController = ng.controller('waitingOrderRegionCon
...
+            await $scope.launchSearch($scope.display.projects, false, false, 0);
             Utils.safeApply($scope);
         };
 
-        $scope.openFiltersLightbox = () => {
-            template.open('lightbox.waitingAdmin', 'administrator/order/filters');
-            $scope.display.lightbox.waitingAdmin = true;
-            Utils.safeApply($scope);
-        }
-
```
Comment (BastienBenard) [inline src/main/resources/public/ts/controllers/administrator/order/waitingOrder.ts:76]: Bloque déplacé dans ProjectFilter

File: src/main/resources/public/ts/controllers/administrator/order/waitingOrder.ts [hunk 6]
```diff
@@ -135,40 +113,6 @@ export const waitingOrderRegionController = ng.controller('waitingOrderRegionCon
...
-                });
-                $scope.filtersFront.all.push(newFilterFront);
-            }
-            if ($scope.filters.all.length > 0) {
-                await $scope.searchProjectAndOrders(false, false, false);
-            } else {
-                await $scope.searchByName($scope.query_name);
-            }
-        };
-
```
Comment (BastienBenard) [inline src/main/resources/public/ts/controllers/administrator/order/waitingOrder.ts:171]: Remplacer par ProjectFilter

File: src/main/resources/public/ts/controllers/administrator/order/waitingOrder.ts [hunk 7]
```diff
@@ -203,17 +147,212 @@ export const waitingOrderRegionController = ng.controller('waitingOrderRegionCon
...
+            await $scope.launchSearch(projects, false, true, null);
+            await projects.exportCSV(false, $scope.display.allOrdersSelected);
+            Utils.safeApply($scope);
+        }
+
+        $scope.loadNextPage = async (): Promise<void> => {
+            $scope.filter.page++;
+            await $scope.launchSearch(new Projects(), true, false, $scope.filter.page);
+            Utils.safeApply($scope);
+        };
```
Comment (BastienBenard) [inline src/main/resources/public/ts/controllers/administrator/order/waitingOrder.ts:161]: Utiliser par le scoll

File: src/main/resources/public/ts/controllers/administrator/order/waitingOrder.ts [hunk 8]
```diff
@@ -203,17 +147,206 @@ export const waitingOrderRegionController = ng.controller('waitingOrderRegionCon
...
+            return totalPrice.toFixed(roundNumber);
+        };
+
+        const calculateAmountRegion = (orders: OrdersRegion) => {
+            let totalAmount = 0;
+            orders.forEach(order => {
+                totalAmount += order.amount;
+            });
+            return totalAmount;
+        };
```
Comment (BastienBenard) [inline src/main/resources/public/ts/controllers/administrator/order/waitingOrder.ts:325]: Fonction reprise de waitingValidatorOrderController. Peut surement être amélioré mais ce sera fait dans une autre refacto.

**Review by sjollois — APPROVED** [review]

**Review by Sofianebernoussi — APPROVED** [review]

**Pull request merged**